torch/_mock_dispatcher/dispatch_key.py

```diff
@@ -0,0 +1,248 @@
+from enum import Enum, auto
+
+class DispatchKey(Enum):
```

Comment:
Sorry it has been a while so I don't remember the context behind this (and I'm still reading this so apologizes if this gets answered somewhere in the PR) -- what's the purpose of adding the Mock Dispatcher? Is it to do parity checking between the PyTorch dispatcher and the Mock Dispatcher?

Reply:
Yeah good question - there were two reasons I made this PR:

(1) It was easier to reason through and debug the changes in python first before porting them to C++
(2) I could imagine it being useful in the future to have a prototype of the dispatcher in python to test out changes in the future.

That said, this python version doesn't map completely to the C++ version - I left out a bunch of features that weren't useful to me, like schema parsing and boxing/unboxing. I also don't have any tests, since I used the python dispatcher mostly just to prove out the design and minimize the time I spent debugging C++.

I probably won't spend time cleaning this up and adding features / tests. So I'd be okay with not landing this PR (and if it would be useful in the future to play around with a python prototype of the dispatcher, I could always grab the changes directly from this PR). But if anyone has stronger feelings about landing / not landing it, I'd be okay either way.